skyve-war/src/main/java/modules/admin/UserLoginRecord/jobs/DifferentCountryLoginNotificationJob.java

```diff
@@ -0,0 +1,76 @@
+package modules.admin.UserLoginRecord.jobs;
+
+import java.util.List;
+import java.util.Locale;
+
+import org.skyve.impl.util.UtilImpl;
+import org.skyve.job.Job;
+import org.skyve.util.CommunicationUtil;
+
+import modules.admin.ModulesUtil;
+import modules.admin.domain.Contact;
+import modules.admin.domain.UserLoginRecord;
+
+/**
+ * This job is used to send an email to a user notifying them that their account was accessed from a different country. It is called
+ * when the system notices a change in the user's country from their last log in.
+ * The job uses a {@code UserLoginRecord} bean parameter that's passed when the scheduler calls the job.
+ */
+public class DifferentCountryLoginNotificationJob extends Job {
+
+	private static final String SUCCESS_MESSAGE = "Successfully sent email warning of unusual login activity to %s";
+
+	public static final String JOB_NAME = "jDifferentCountryLoginNotification";
+
+	private static final String UNUSUAL_LOGIN_ACTIVITY_EMAIL_DEFAULT_SUBJECT = "Security Alert: Unusual Login Activity Detected";
+
+	private static final String UNUSUAL_LOGIN_EMAIL_BODY = "Dear %s,\n\n"
+			+ "We have detected a login attempt to your account from a new location: %s with IP address: %s. "
+			+ "If this was you, there's no need to take further action.\n\n"
+			+ "However, if you do not recognize this activity, we strongly recommend that you change your password immediately to secure your account. "
+			+ "You can do this by logging into your account and navigating to <strong>Admin -> Password</strong>.\n\n"
+			+ "For your safety, please do not share your password with anyone.\n\n"
+			+ "If you have any questions or need assistance, please contact our support team at %s.\n\n"
+			+ "Best regards,\n"
+			+ "The Security Team";
+
+	private static final String UNUSUAL_LOGIN_EMAIL_FAILURE_LOG_MESSAGE = "Failed to send security alert email to %s at %s regarding a login from a different country. Exception: %s";
+
+	public static final String COMMUNICATION_DESCRIPTION = "Email warning a user of a new login from a different country";
+
+	@Override
+	public void execute() throws Exception {
+		List<String> log = getLog();
+		setPercentComplete(0);
+
+		UserLoginRecord loginRecord = (UserLoginRecord) this.getBean();
+		String userIPAddress = loginRecord.getIpAddress();
+		String countryCode = loginRecord.getCountryCode();
+
+		if (countryCode != null) {
+			// Get actual country name
+			Locale locale = new Locale("", countryCode);
+			String country = locale.getDisplayCountry();
+
+			Contact contact = ModulesUtil.getCurrentUserContact();
+			String userEmail = contact.getEmail1();
+			String userName = contact.getName();
+			try {
+				CommunicationUtil.sendFailSafeSystemCommunication(COMMUNICATION_DESCRIPTION,
+						UNUSUAL_LOGIN_ACTIVITY_EMAIL_DEFAULT_SUBJECT,
+						String.format(UNUSUAL_LOGIN_EMAIL_BODY, userName, country, userIPAddress, UtilImpl.SUPPORT_EMAIL_ADDRESS),
```

Comment:
Do not use `String.format` with a `Communication`. It will not be substituted at runtime, it only affects the _first_ template that is created. If you want dynamic content, you need to use Skyve bindings (`{}`) which will be substituted at runtime during each send.